### ISSUE TITLE
F-Spot source icon image loading error

### ISSUE THREAD

**Issue by GoogleCodeExporter** [yendo/gphotoframe#50]
```
(from issue #49)

> gconftool-2 --type bool --set /apps/gphotoframe/window_sticky 0
> ./gphotoframe
> [ right click, choose preferences ]
> Warning: the desktop entry of gphotoframe is not found.
> Traceback (most recent call last):
>   File "/home/gwhite/Download/gphotoframe-0.7/lib/preferences.py", line
42, in start
>     gui, "treeview2", plugins.PluginListStore(), self.prefs)
>   File "/home/gwhite/Download/gphotoframe-0.7/lib/plugins/__init__.py",
line 47, in 
> __init__
>     list = [ obj.is_available(), obj.get_icon_pixbuf(), name ]
>   File "/home/gwhite/Download/gphotoframe-0.7/lib/plugins/base.py", line
24, in 
> get_icon_pixbuf
>     pixbuf = self.icon().get_pixbuf()
>   File "/home/gwhite/Download/gphotoframe-0.7/lib/utils/iconimage.py",
line 27, in 
> get_pixbuf
>     pixbuf = gtk.gdk.pixbuf_new_from_file(file)
> TypeError: pixbuf_new_from_file() argument 1 must be string, not None

```

Original issue reported on code.google.com by `yendo0206` on 6 May 2010 at 2:01

**Comment by GoogleCodeExporter** [comment]
```
Have you installed the gnome-icon-theme?
Please teach the result of the following commands. 

locate image-x-generic
```

Original comment by `yendo0206` on 6 May 2010 at 2:09

**Comment by GoogleCodeExporter** [comment]
```
I have Gnome Icon Theme 2.28.0 as installed via Fedora 12's RPM 
gnome-icon-theme-
2.28.0-1.fc12.noarch

# locate image-x-generic | sort
/usr/share/icons/gnome/16x16/mimetypes/image-x-generic.png
/usr/share/icons/gnome/22x22/mimetypes/image-x-generic.png
/usr/share/icons/gnome/24x24/mimetypes/image-x-generic.png
/usr/share/icons/gnome/32x32/mimetypes/image-x-generic.png
/usr/share/icons/gnome/scalable/mimetypes/image-x-generic.svg
/usr/share/icons/HighContrastLargePrint/48x48/mimetypes/image-x-generic.png
/usr/share/icons/HighContrastLargePrintInverse/48x48/mimetypes/image-x-generic.p
ng
/usr/share/icons/HighContrast-SVG/scalable/mimetypes/image-x-generic.svg
/usr/share/icons/nuoveXT2/128x128/mimetypes/image-x-generic.png
/usr/share/icons/nuoveXT2/16x16/mimetypes/image-x-generic.png
/usr/share/icons/nuoveXT2/22x22/mimetypes/image-x-generic.png
/usr/share/icons/nuoveXT2/24x24/mimetypes/image-x-generic.png
/usr/share/icons/nuoveXT2/32x32/mimetypes/image-x-generic.png
/usr/share/icons/nuoveXT2/36x36/mimetypes/image-x-generic.png
/usr/share/icons/nuoveXT2/48x48/mimetypes/image-x-generic.png
/usr/share/icons/nuoveXT2/64x64/mimetypes/image-x-generic.png
/usr/share/icons/nuoveXT2/72x72/mimetypes/image-x-generic.png
/usr/share/icons/nuoveXT2/96x96/mimetypes/image-x-generic.png
/usr/share/icons/oxygen/128x128/mimetypes/image-x-generic.png
/usr/share/icons/oxygen/16x16/mimetypes/image-x-generic.png
/usr/share/icons/oxygen/22x22/mimetypes/image-x-generic.png
/usr/share/icons/oxygen/256x256/mimetypes/image-x-generic.png
/usr/share/icons/oxygen/32x32/mimetypes/image-x-generic.png
/usr/share/icons/oxygen/48x48/mimetypes/image-x-generic.png
/usr/share/icons/oxygen/64x64/mimetypes/image-x-generic.png
```

Original comment by `graham_a...@hotmail.com` on 6 May 2010 at 3:15

**Comment by GoogleCodeExporter** [comment]
```
Thanks for your help.

I attach gphotoframe debug version.
Could you try it?
Please teach the result of the following commands. 

# Have you installed f-spot?

```

Original comment by `yendo0206` on 6 May 2010 at 3:49

Attachments:
- [gphotoframe-0.7.1-b1.tar.gz](https://storage.googleapis.com/google-code-attachments/gphotoframe/issue-50/comment-3/gphotoframe-0.7.1-b1.tar.gz)

**Comment by GoogleCodeExporter** [comment]
```
I don't have f-spot installed.

Here's the result of running 0.7.1-b1 and choosing "preferences"...

./gphotoframe
Warning: the desktop entry of gphotoframe is not found.
f-spot
Traceback (most recent call last):
  File "/home/gwhite/Download/gphotoframe-0.7.1-b1/lib/preferences.py", line 42, in 
start
    gui, "treeview2", plugins.PluginListStore(), self.prefs)
  File "/home/gwhite/Download/gphotoframe-0.7.1-b1/lib/plugins/__init__.py", line 47, 
in __init__
    list = [ obj.is_available(), obj.get_icon_pixbuf(), name ]
  File "/home/gwhite/Download/gphotoframe-0.7.1-b1/lib/plugins/base.py", line 24, in 
get_icon_pixbuf
    pixbuf = self.icon().get_pixbuf()
  File "/home/gwhite/Download/gphotoframe-0.7.1-b1/lib/utils/iconimage.py", line 28, 
in get_pixbuf
    pixbuf = gtk.gdk.pixbuf_new_from_file(file)
TypeError: pixbuf_new_from_file() argument 1 must be string, not None
```

Original comment by `graham_a...@hotmail.com` on 7 May 2010 at 12:41

**Comment by GoogleCodeExporter** [comment]
```
Thank you.

> ./gphotoframe
> Warning: the desktop entry of gphotoframe is not found.
> f-spot

It is a F-spot icon loading bug.
I fixed the bug and attach the new version (0.7.1-b2).
Could you try it?

```

Original comment by `yendo0206` on 7 May 2010 at 3:20
- Changed title: **F-Spot source icon image loading error**
- Changed state: **Started**

Attachments:
- [gphotoframe-0.7.1-b2.tar.gz](https://storage.googleapis.com/google-code-attachments/gphotoframe/issue-50/comment-5/gphotoframe-0.7.1-b2.tar.gz)

**Comment by GoogleCodeExporter** [comment]
```
0.7.1-b2 seems to work well, no errors - thanks!
```

Original comment by `graham_a...@hotmail.com` on 7 May 2010 at 7:34

**Comment by GoogleCodeExporter** [comment]
```
Thank you for confirmation.
Fixed in revision 0c3f81679c.
```

Original comment by `yendo0206` on 8 May 2010 at 4:39
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Issue 51 has been merged into this issue.
```

Original comment by `yendo0206` on 8 May 2010 at 6:33
